Backend/templates/gitlab-ci/master-gitlab-ci.yml

```diff
@@ -0,0 +1,274 @@
+image: alpine:3.9.3
+
+
+variables:
+  HELM_DOCKER_TAG: "v2.14.0"
+  GSG_DOCKER_TAG: "v0.20.4"
+
+
+  OPEN_API_DOCKER_VERSION: "v4.0.1"
+  GSG_TAG_PREFIX: "releases/"
+
+
+stages:
+  - MergeRequest
+  - BuildNumber
+  - Build
+  - Lint
+  - Docker
+  - Documentation
+  - TagBuildNumber
+  - TagVersion
+  - Deploy
+
+
+##############
+# Merge jobs #
+##############
+
+.build_mr_template:
+  stage: MergeRequest
+  only:
+    - merge_requests
+  
+
+.lint_mr_template:
+  stage: MergeRequest
+  only:
+    - merge_requests
+
+
+.lint_helm_mr_template:
+  image: devth/helm
+  stage: MergeRequest
+  only:
+    - merge_requests
+  script:
+    - helm lint $CHART_NAME
+
+
+.test_mr_template:
+  stage: MergeRequest
+  only:
+    - merge_requests
+  script:
+    - echo 'Tests ✅' 
+
+
+###############
+# Master jobs #
+###############
+
+BuildNumber:
+  stage: BuildNumber
+  image: samueldebruyn/debian-git
+  only:
+    - master
+  script:
+    - git rev-list --count origin/master > .build_number
+    - echo "Build number:"$(cat .build_number)
+  artifacts:
+    paths:
+      - .build_number
+    
+
+Docker:
+  stage: Docker
+  image: docker:stable
+  variables:
+    DOCKER_HOST: tcp://docker:2375/
+    DOCKER_DRIVER: overlay2
+  services:
+    - docker:dind
+  only:
+    - master
+  script:
+    - BUILD_NUMBER=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+    - docker build -t $CI_REGISTRY/$CI_PROJECT_PATH/builds:$BUILD_NUMBER .
+    - docker push $CI_REGISTRY/$CI_PROJECT_PATH/builds:$BUILD_NUMBER
+
+
+TagBuildNumber:
+  stage: TagBuildNumber
+  image: samueldebruyn/debian-git
+  only:
+    - master
+  script:
+    - BUILD_NUMBER=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+    
+    - git tag builds/$BUILD_NUMBER
+    - git remote set-url origin https://gitlab-ci-token:$GL_TOKEN@gitlab.com/$CI_PROJECT_PATH.git/
+    - git push origin --tags
+
+
+TagVersion:
+  stage: TagVersion
+  image: registry.gitlab.com/juhani/go-semrel-gitlab:$GSG_DOCKER_TAG
+  when: manual
+  only:
+    - master
+  script:
+    - release changelog
+    - release commit-and-tag CHANGELOG.md --create-tag-pipeline
+
+
+
+##################
+# Build tag jobs #
+##################
+
+DeployVersionDev: 
+  stage: BuildNumber
+  only:
+    - /builds/
+  except:
+    - branches
+  script:
+    - echo $CI_COMMIT_TAG | sed -n 's/builds\/\([0-9]\)/\1/p' > .build_number
+    - echo "Build number "$(cat .build_number)
+  artifacts:
+    paths:
+      #TODO maybe this should be a different name? 🤔
+      - .build_number 
+
+
+DeployDev:
+  extends: .deploy_template
+  variables:
+    PROJECT: "youclap-dev"
```

Comment:
this could probably be an environment variable, no?

Reply:
Maybe in the BE group? 🤔